piptools/scripts/compile.py

```diff
@@ -383,9 +383,10 @@ def cli(
 
     if resolver_name == "legacy":
         log.warning(
-            "WARNING: using legacy resolver is deprecated and will be removed in "
-            "future versions. The default resolver will be change to 'backtracking' "
-            "in 7.0.0 version. Specify --resolver=backtracking to silence this warning."
+            "WARNING: the legacy dependency resolver is deprecated and will be removed"
```

Comment:
Since the message is changed this would require updating the corresponding test.